### DISCUSSION TITLE
How to describe a RegExp object?

### DISCUSSION THREAD

**Discussion by maddsua** [colinhacks/zod#2753] [Q&A] [answered]
So I'm building a config schema for my tool, and I'd like it to have a regex option for pattern matching.

This is the type I'm trying to get from Zod schema:

```typescript
interface Config {
  ...

  exclude: (string | RegExp)[]

  ...
}
```

I'm using Zod for like two days now, so my experience with it is near zero. My google searches did not return any meaningful results for my usecase, so help would be appreciated

**Comment by maddsua** [comment]
For now my solution is to define schema like this:

```typescript
...
  exclude: z.array(z.string().or(z.any())),
...
```

then redefine the type with:

```typescript
export interface ConfigSchema extends z.infer<typeof configSchema> {
...
  exclude: (string | RegExp)[];
...
};
```

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const schema = z.object( {
    exclude: z.string().or( z.instanceof( RegExp ) ).array(),
} )
type Data = z.infer<typeof schema>
// type Data = {
//     exclude: ( string | RegExp )[]
// }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by maddsua** [reply]
Oh, zod had instanceof all that time. Yes, that's definitely what I was looking for, thanks!
